lib/bolt_server/transport_app.rb

```diff
@@ -135,11 +135,30 @@ def task_helper(target, task, parameters)
         end
       end
 
-      @executor.run_task(target, task, parameters).each do |result|
-        value = result.value
-        next unless value.is_a?(Hash)
-        next unless value.key?('_sensitive')
-        value['_sensitive'] = value['_sensitive'].unwrap
+      if timeout
+        task_thread = Thread.new do
+          @executor.run_task(target, task, parameters).each do |result|
```

Comment:
This copy-paste needs to be cleaned up. We also need to account for the case where the timeout is `0`.

Reply:
Oh, right I forgot about the special 0 case. Updated.